mtags/src/main/scala-2/scala/meta/internal/pc/PcRenameProvider.scala

```diff
@@ -14,15 +14,26 @@ class PcRenameProvider(
   private val forbiddenMethods =
     Set("equals", "hashCode", "unapply", "unary_!", "!")
 
+  private val soughtSymbolNames = soughtSymbols match {
+    case Some((symbols, _)) =>
+      symbols
+        .filterNot(_.isErroneous)
+        .map(symbol => symbol.decodedName.toString)
+    case None => Set.empty[String]
+  }
+
   def canRenameSymbol(sym: Symbol): Boolean = {
-    (!sym.isMethod || !forbiddenMethods(sym.decodedName)) &&
-    (sym.ownersIterator
-      .drop(1)
-      .exists(owner =>
-        owner.isMethod || owner.isAnonymousFunction
-      )) // this also works for worksheets, since they are wrapped in `method main`
+    val name = sym.decodedName.toString
+    def sameName = soughtSymbolNames(name)
```

Comment:
How does this work with e.g `apply` or `<init>`?

Reply:
Good catch, I added them to forbidden methods. We could work something out with apply, but I don't really want to tackle it now and it's a rare enough scenario.